src/components/NewVaultOfferSummary.tsx

```diff
@@ -14,8 +26,93 @@ const TableRow = ({ left, right }: TableRowProps) => {
   );
 };
 
-const NewVaultOfferSummary = () => {
-  const isDisabled = true;
+type Props = {
+  inputErrors: VaultCreationErrors;
+};
+
+const NewVaultOfferSummary = ({ inputErrors }: Props) => {
+  const { collateralizationRatioError, toLockError, toReceiveError } =
+    inputErrors;
+
+  const selectedCollateralId = useAtomValue(selectedCollateralIdAtom);
+  const valueToReceive = useAtomValue(valueToReceiveAtom);
+  const valueToLock = useAtomValue(valueToLockAtom);
+  const collateralizationRatio = useAtomValue(collateralizationRatioAtom);
+  const offerSigner = useAtomValue(offerSignerAtom);
+
+  const { displayAmount, displayBrandPetname, displayPercent } =
+    useAtomValue(displayFunctionsAtom) ?? {};
+
+  const { metrics, params, factoryParams } = useVaultStore(vaults => ({
+    metrics: vaults.vaultMetrics,
+    params: vaults.vaultGovernedParams,
+    factoryParams: vaults.vaultFactoryParams,
+  }));
+
+  const selectedParams =
+    selectedCollateralId && params.get(selectedCollateralId);
+
+  const selectedMetrics =
+    selectedCollateralId && metrics.get(selectedCollateralId);
+
+  const collateralBrand =
+    selectedMetrics && selectedMetrics.totalCollateral.brand;
+  const debtBrand = selectedMetrics && selectedMetrics.totalDebt.brand;
+
+  const depositAmount =
+    valueToLock &&
+    collateralBrand &&
+    AmountMath.make(collateralBrand, valueToLock);
+
+  const depositAmountForDisplay =
+    displayAmount && displayBrandPetname && depositAmount
+      ? `${displayAmount(depositAmount, 2)} ${displayBrandPetname(
+          collateralBrand,
+        )}`
+      : '--';
+
+  const borrowAmount =
+    valueToReceive && debtBrand && AmountMath.make(debtBrand, valueToReceive);
+
+  const borrowAmountForDisplay =
+    displayAmount && displayBrandPetname && borrowAmount
+      ? `${displayAmount(borrowAmount, 2)} ${displayBrandPetname(debtBrand)}`
+      : '--';
+
+  const interestRateForDisplay =
+    displayPercent && selectedParams
+      ? `${displayPercent(selectedParams.interestRate, 0)}%`
+      : '--';
+
+  const creationFeeForDisplay =
+    displayPercent && selectedParams
+      ? `${displayPercent(selectedParams.loanFee, 0)}%`
+      : '--';
+
+  const liquidationRatioForDisplay =
+    displayPercent && selectedParams
+      ? `${displayPercent(selectedParams.liquidationMargin, 0)}%`
+      : '--';
+
+  const collateralizationRatioForDisplay =
+    displayPercent && collateralizationRatio
+      ? `${displayPercent(collateralizationRatio, 0)}%`
+      : '--';
+
+  const isDisabled =
```

Comment:
all these bangs. ditto re hasPriceFeed for boolean handling and also this makes more sense as an `&&`. All these conditions must be met.

Above you name the state semantically like `isInputReady` and then map that to the UI concern of "disabled" in that property. I think that's a better way to go here too.

```
  const hasErrors = collateralizationRatioError || toLockError || toReceiveError;
  const canCreateVault: boolean = 
    !hasErrors &&
    selectedMetrics && 
    selectedParams && 
```

Reply:
Done